designs/disruption-controls-by-reason.md

```diff
@@ -0,0 +1,365 @@
+# Disruption Controls By Reason
+# Table of Contents
+1. [Disruption Controls By Reason](#disruption-controls-by-reason)
+2. [API Design](#api-design)
+   1. [Approach A: Extending the Budget API to Specify a Reason](#approach-a-extending-the-budget-api-to-specify-a-reason)
+      1. [List Approach: Multiple Reasons per Budget - Recommended](#list-approach-multiple-reasons-per-budget---recommended)
+      2. [Single Reason Approach: One Reason per Budget](#single-reason-approach-one-reason-per-budget)
+   2. [Approach B: Defining Per Reason Controls](#approach-b-defining-per-reason-controls)
+   3. [Preferred Option: List Approach](#preferred-option-list-approach)
+   4. [Disruption Behavior](#disruption-behavior)
+   5. [Scenarios](#scenarios)
```

Comment:
```suggestion
   5. [User Stories](#scenarios)
```

This is a nit, but most KEPs call them user stories, make sure the header below is aligned with this naming

Reply:
I was wondering about this. Should we be using the KEP template now that we are part of the kubernetes project proper?